### PR TITLE
Remove support for file to opt-out of telemetry, only support env var

### LINKED ISSUE #6078: remove file opt-out from telemetery and only use environment variable
We should only have one opt-out mechanism.  We should remove the file opt-out as it is not flexible enough for all situations.

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
## PR Summary

Since a PR added support to opt out of telemetry via an environment variable, we can remove the
always intended to be temporary solution of deleting a file to opt out of telemetry since the
environment variable can be defined at the system level and exist before even installing PowerShell Core.

Because the variable is defined as `opt out`, a value of `true`, `yes`, or `1` means no telemetry is sent.

Fix https://github.com/PowerShell/PowerShell/issues/6078

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [X] User-facing https://github.com/PowerShell/PowerShell-Docs/pull/2316
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [x] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/Telemetry.cs [hunk 1]
```diff
@@ -76,7 +67,7 @@ private static void SendTelemetry(string eventName, Dictionary<string,string>pay
             try
             {
                 // if the semaphore file exists, try to send telemetry
```
Comment (adityapatwardhan) [inline src/Microsoft.PowerShell.ConsoleHost/host/msh/Telemetry.cs]: Please update the comment.

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/Telemetry.cs [hunk 2]
```diff
@@ -76,7 +67,7 @@ private static void SendTelemetry(string eventName, Dictionary<string,string>pay
...
-                var enabled = Utils.NativeFileExists(TelemetrySemaphoreFilePath) && !GetEnvironmentVariableAsBool(TelemetryOptoutEnvVar, false);
+                var enabled = GetEnvironmentVariableAsBool(TelemetryOptoutEnvVar, false);
```
Comment (adityapatwardhan) [inline src/Microsoft.PowerShell.ConsoleHost/host/msh/Telemetry.cs]: Please use named arguments.

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

File: tools/appveyor.psm1 [hunk 3]
```diff
@@ -347,11 +347,8 @@ function Invoke-AppVeyorTest
         Write-Host -Foreground Green 'Running all CoreCLR tests..'
     }
 
-    # Remove telemetry semaphore file in CI
-    $telemetrySemaphoreFilepath = Join-Path $env:CoreOutput DELETE_ME_TO_DISABLE_CONSOLEHOST_TELEMETRY
-    if ( Test-Path "${telemetrySemaphoreFilepath}" ) {
-        Remove-Item -Force ${telemetrySemaphoreFilepath}
-    }
+    #  Disable telemetry in CI
+    $env:POWERSHELL_TELEMETRY_OPTOUT = 1
```
Comment (TravisEz13) [inline tools/appveyor.psm1]: Why not add this in `Start-PSPester`?

**Review by TravisEz13 — DISMISSED** [review]

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/Telemetry.cs (hunk 2 above)
Comment (daxian-dbw) [inline src/Microsoft.PowerShell.ConsoleHost/host/msh/Telemetry.cs]: This is not right. You are returning `false` by default (when the env variable is not set), and thus we will not send telemetry by default. You should pass `true` as the default value.

File: tools/appveyor.psm1 (hunk 3 above)
Comment (daxian-dbw) [inline tools/appveyor.psm1]: Setting 1 to the env variable means you enable the telemetry. You should set 0 here.

File: tools/travis.ps1 [hunk 4]
```diff
@@ -224,11 +224,8 @@ elseif($Stage -eq 'Build')
         $pesterParam['IncludeFailingTest'] = $true
     }
 
-    # Remove telemetry semaphore file in CI
-    $telemetrySemaphoreFilepath = Join-Path $output DELETE_ME_TO_DISABLE_CONSOLEHOST_TELEMETRY
-    if ( Test-Path "${telemetrySemaphoreFilepath}" ) {
-        Remove-Item -force ${telemetrySemaphoreFilepath}
-    }
+    # Disable telemetry in CI
+    $env:POWERSHELL_TELEMETRY_OPTOUT = 1
```
Comment (daxian-dbw) [inline tools/travis.ps1]: Same here, you should set 0 here.

**Review by daxian-dbw — CHANGES_REQUESTED** [review]

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/Telemetry.cs (hunk 2 above)
Comment (daxian-dbw) [reply src/Microsoft.PowerShell.ConsoleHost/host/msh/Telemetry.cs]: `POWERSHELL_TELEMETRY_OPTOUT = 1` or `POWERSHELL_TELEMETRY_OPTOUT = yes` means to not send telemetry, while `POWERSHELL_TELEMETRY_OPTOUT = 0` or `POWERSHELL_TELEMETRY_OPTOUT = no` means to send telemetry?

This looks so not intuitive to me. When I set `POWERSHELL_TELEMETRY_OPTOUT = no`, I would expect it means no telemetry, literally.

Comment (daxian-dbw) [reply src/Microsoft.PowerShell.ConsoleHost/host/msh/Telemetry.cs]: My bad, I misread POWERSHELL_TELEMETRY_OPTOUT to POWERSHELL_TELEMETRY_**OUTPUT** ...

**Review by daxian-dbw — APPROVED** [review]

**Comment by TravisEz13** [comment]
I've tried to figure out why travis-ci is failing.  I'm going to close an re-open to PR to see if that resolves the issue.

**Review by TravisEz13 — DISMISSED** [review]
The YML change seems to have broken Travis-CI

**Comment by TravisEz13** [comment]
I added a fix to the YML

**Comment by SteveL-MSFT** [comment]
@TravisEz13 @adityapatwardhan your feedback has been addressed

**Review by adityapatwardhan — APPROVED** [review]

**Pull request merged**